### PR TITLE
[add] cms_notification

### PR DESCRIPTION
**Pull request by simahawk**
CMS form for managing personal notifications settings.

Depends on [mail_digest PR](https://github.com/OCA/social/pull/158) and [cms_form PR](https://github.com/OCA/website-cms/pull/9)

### CODE REVIEW

**Comment by simahawk** [comment]
TODO: fix PR dependencies.

File: cms_notifications/models/res_users.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        msg_model = self.env['mail.message']
+        for item in self:
+            if not item.partner_id:
+                continue
+            domain = [
+                ('partner_ids', 'in', item.partner_id.id),
+                ('needaction_partner_ids', 'in', item.partner_id.id),
+                ('subtype_id.cms_type', '=', True),
+            ]
+            item.has_unread_notif = bool(msg_model.search_count(domain))
```
Comment (guewen) [inline cms_notifications/models/res_users.py]: With a large number of `mail.message`, a `search` with a `limit=1` is drastically faster than a `search_count`. It has to search only 1 line vs all of them.

**Review by guewen — COMMENTED** [review]
Performance improvement suggestion.
I like the feature!

**Comment by pedrobaeza** [comment]
Please don't use plurals on module name. cms_notification is good enough I think, or if not, cms_multi_notification.

**Comment by simahawk** [comment]
@pedrobaeza is it that fundamental? I have it already in production and I'd love to skip a painful migration here :stuck_out_tongue:

**Comment by pedrobaeza** [comment]
Yeah, I think so. It's already a guideline and avoid typos. You can migrate it very easily with openupgradelib and shell command:
```
from openupgradelib import openupgrade

openupgrade.update_module_names([('cms_notifications', 'cms_notification')])
```

**Comment by simahawk** [comment]
@pedrobaeza didn't knew that. TIL thanks :)
I'll rename and squash.

**Comment by pedrobaeza** [comment]
EDIT: I have fixed the syntax of the code that was incorrect

**Comment by simahawk** [comment]
You can hit this button too after OCA/social/pull/158 is merged :heart:

**Comment by simahawk** [comment]
@pedrobaeza OCA/social#158 merged, any blocker here?

File: cms_notification/README.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        @property
+        def _form_subtype_fields(self):
+            res = super(CMSNotificationPanel, self)._form_subtype_fields
+            res.update({
+                'enable_my_type': 'module.mt_mytype',
+                'enable_my_type2': 'module.mt_mytype2',
+            })
+            return res
+
+            Usage
```
Comment (pedrobaeza) [inline cms_notification/README.rst]: This is incorrectly indented

File: cms_notification/i18n/cms_notifications.pot [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline cms_notification/i18n/cms_notifications.pot]: Remove POT file

File: oca_dependencies.txt [hunk 4]
```diff
@@ -1,15 +1 @@
...
-# To depend on the standard version of sale-workflow, use:
-# sale-workflow
-#
-# To explicitely give the URL of a fork, and still use the version specified in
-# .travis.yml, use:
-# sale-workflow https://github.com/OCA/sale-workflow
-#
-# To provide both the URL and a branch, use:
-# sale-workflow https://github.com/OCA/sale-workflow branchname
+social https://github.com/camptocamp/social add-mail_digest
```
Comment (pedrobaeza) [inline oca_dependencies.txt]: This should be changed

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Why don't you use FontAwesome included in Odoo?

**Comment by simahawk** [comment]
@pedrobaeza font-awesome from odoo is old, at least in v9

**Comment by pedrobaeza** [comment]
Yeah, but possibly the icons (or similar) used here are in this version, because bundling libraries on each module is very heavy, and more if the use is very light as this one.

**Comment by simahawk** [comment]
@pedrobaeza they are not. There's no envelope open for instance. This is why I added it. To be honest: I'd say that icons should be updated in OCB. Here, maybe, I can create a custom one...

**Comment by pedrobaeza** [comment]
OK, what about having a module base_fontawesome with this lib? On v10, we can deprecate it as the version is more modern. Can you obtain the same way of declaring icons with this new module, so you don't have to change anything in this module?

**Comment by simahawk** [comment]
hmm could be a good idea and allow to reuse it for several v9 modules.

**Comment by simahawk** [comment]
@pedrobaeza there you go https://github.com/OCA/server-tools/pull/866

**Comment by simahawk** [comment]
final bits to get this merged: replace oca dependency url as OCA/server-tools#866 has been merged a long time ago.

**Pull request merged**